From 824b8e8e5f6ffa277133d19449f83e0589bbcc92 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <Michal@Hrusecky.net>
Date: Thu, 18 Jan 2018 13:19:06 +0100
Subject: [PATCH] python: Fix build

In installation ffi library couldn't be found, adding LD_LIBRARY_PATH fixes it.

Signed-off-by: Michal Hrusecky <Michal@Hrusecky.net>
---
 lang/python/python-host.mk   | 1 +
 lang/python/python/Makefile  | 1 +
 lang/python/python3-host.mk  | 1 +
 lang/python/python3/Makefile | 1 +
 4 files changed, 4 insertions(+)

diff --git a/lang/python/python-host.mk b/lang/python/python-host.mk
index d4dc81f..b271fd6 100644
--- a/lang/python/python-host.mk
+++ b/lang/python/python-host.mk
@@ -36,6 +36,7 @@ define HostPython
 		export _python_prefix="/usr"; \
 		export _python_exec_prefix="/usr"; \
 	fi; \
+	export LD_LIBRARY_PATH="$$(STAGING_DIR_HOST)/lib:$$(STAGING_DIR_HOSTPKG)/lib"; \
 	export PYTHONOPTIMIZE=""; \
 	$(1) \
 	$(HOST_PYTHON_BIN) $(2);
diff --git a/lang/python/python/Makefile b/lang/python/python/Makefile
index c70c486..58262d7 100644
--- a/lang/python/python/Makefile
+++ b/lang/python/python/Makefile
@@ -287,6 +287,7 @@ define Host/Compile
 endef
 
 define Host/Install
+	+LD_LIBRARY_PATH="$$(STAGING_DIR_HOST)/lib:$$(STAGING_DIR_HOSTPKG)/lib" \
 	$(MAKE) -C $(HOST_BUILD_DIR) install
 	$(INSTALL_DIR) $(HOST_PYTHON_DIR)/bin/
 	$(INSTALL_BIN) $(HOST_BUILD_DIR)/Parser/pgen $(HOST_PYTHON_DIR)/bin/pgen2
diff --git a/lang/python/python3-host.mk b/lang/python/python3-host.mk
index 2a418d5..e545b96 100644
--- a/lang/python/python3-host.mk
+++ b/lang/python/python3-host.mk
@@ -36,6 +36,7 @@ define HostPython3
 		export _python_prefix="/usr"; \
 		export _python_exec_prefix="/usr"; \
 	fi; \
+	export LD_LIBRARY_PATH="$$(STAGING_DIR_HOST)/lib:$$(STAGING_DIR_HOSTPKG)/lib"; \
 	export PYTHONOPTIMIZE=""; \
 	$(1) \
 	$(HOST_PYTHON3_BIN) $(2);
diff --git a/lang/python/python3/Makefile b/lang/python/python3/Makefile
index cd97378..329092f 100644
--- a/lang/python/python3/Makefile
+++ b/lang/python/python3/Makefile
@@ -284,6 +284,7 @@ define Host/Compile
 endef
 
 define Host/Install
+	+LD_LIBRARY_PATH="$$(STAGING_DIR_HOST)/lib:$$(STAGING_DIR_HOSTPKG)/lib" \
 	$(MAKE) -C $(HOST_BUILD_DIR) install
 	$(INSTALL_DIR) $(HOST_PYTHON3_DIR)/bin/
 	$(INSTALL_BIN) $(HOST_BUILD_DIR)/Parser/pgen $(HOST_PYTHON3_DIR)/bin/pgen3
-- 
2.15.1

